src/java.base/share/classes/jdk/internal/classfile/impl/RawBytecodeHelper.java

```diff
@@ -443,12 +442,11 @@ public boolean next() {
         int len = LENGTHS[code & 0xFF]; // & 0xFF eliminates bound check
         this.bci = bci;
         opcode = code;
-        isWide = false;
         if (len <= 0) {
             len = checkSpecialInstruction(bci, end, code); // sets opcode
         }
 
-        if (len <= 0 || (nextBci += len) > end) {
+        if ((nextBci += len) > end) {
```

Comment:
This change makes it that `nextBci` will no longer monotonically increase in case of a malformed special instruction.

```suggestion
        if (len <= 0 || (nextBci += len) > end) {
```

Reply:
`nextBci` is used only if we proceed. I believe none of the use sites processing untrusted bytecode proceeds when it receives an `INVALID` result, so change to `nextBci` in that case has no impact.